src/repr/type_pp.ml

```diff
@@ -149,8 +156,16 @@ let ty : type a. a t Fmt.t =
         | _ ->
             let var = Var (get_tvar ()) in
             Fmt.pf ppf "@[(%a as %a)@]" ty (self_unroll var) ty var)
-    | Boxed b -> Fmt.pf ppf "@[Boxed (%a)@]" ty b
-    | Custom c -> Fmt.pf ppf "@[Custom (%a)@]" custom c
+    | Custom c | Attributes { attr_type = Custom c; _ } ->
```

Comment:
True, but going in the other direction means that the user can't ever directly build just a `Custom ( )`, due to the JSON functions being attached as attributes. I'm not sure there's an ideal solution here.

Reply:
I'm not sure to understand your answer here :)

Reply:
I picked this implementation because otherwise the tests [here](https://github.com/CraigFe/repr/blob/type-attributes/test/repr/main.ml#L575-L577) require adding `Attributes (... json ..., Custom (...))` instead, which leaks the asymmetry of implementation between JSON and non-JSON generic functions. As you point out, this already exists for `Repr.random`, but this is part of a larger inconsistency.

Reply:
I see. Thanks